frontend/src/tests/application-tests.js

```diff
@@ -1,24 +1,391 @@
+        let resp = await apiPOST(
+            `/debug/active_user`,
+            userCreatedFromApplicant
+        );
+        expect(resp).toHaveStatus("success");
+    }
+
+    /**
+     * Switches the current active user to a user with only the ta role.
+     * This function uses debug route to achieve user switching.
+     *
+     * @returns {Promise<void>}
+     */
+    async function switchToTaUser() {
+        let resp = await apiPOST(`/debug/active_user`, userWithTaPermissions);
+        expect(resp).toHaveStatus("success");
+    }
+
+    /**
+     * Restores the active user to the default user (the user logged during test setup in beforeAll).
+     *
+     * @returns {Promise<void>}
+     */
+    async function restoreDefaultUser() {
+        let respSwitchBackUser = await apiPOST(
+            `/debug/active_user`,
+            defaultUser
+        );
+        expect(respSwitchBackUser).toHaveStatus("success");
+    }
+
     beforeAll(async () => {
         await databaseSeeder.seed({ apiGET, apiPOST });
+        session = databaseSeeder.seededData.session;
+        applicant = databaseSeeder.seededData.applicant;
+        position = databaseSeeder.seededData.position;
+
+        // Get the current active user
+        let resp = await apiGET("/debug/active_user");
+        expect(resp).toHaveStatus("success");
+        defaultUser = resp.payload;
+
+        // Making sure the user is the seeded applicant before this get request for the purposes of the next test
+        // Created a user for the applicant to make sure they exist before changing the default user to applicant
+        Object.assign(userCreatedFromApplicant, {
+            utorid: applicant.utorid,
+            roles: ["ta"],
+        });
+
+        resp = await apiPOST("/debug/users", userCreatedFromApplicant);
+        expect(resp).toHaveStatus("success");
+        Object.assign(userCreatedFromApplicant, resp.payload);
+
+        // Create a new user with TA permissions
+        resp = await apiPOST("/debug/users", userWithTaPermissions);
+        expect(resp).toHaveStatus("success");
+        Object.assign(userWithTaPermissions, resp.payload);
     }, 30000);
 
     // These tests set data through the `/public/postings` route,
     // but read data through the `/api/v1/admin` route.
     describe("Public route tests", () => {
-        it.todo("Get survey.js posting data through public route");
+        it("Get survey.js posting data through public route", async () => {
+            // Make sure the user has admin permissions before this post request
+            // Make a new posting and update <posting> to include the id of the posting
+            let resp = await apiPOST("/admin/postings", {
+                ...posting,
+                session_id: session.id,
+            });
+            expect(resp).toHaveStatus("success");
+            checkPropTypes(postingPropTypes, resp.payload);
+            expect(resp.payload.id).not.toBeNull();
```

Comment:
I am not sure this will catch the case when id is simply not returned (i.e. it is `undefined`). Maybe `toBeFalsy()` or `toBeDefined()` is a better check?

Reply:
checkPropTypes already assures that the payload has an id. So all I am checking here is if it is not null.